api/t8_fortran_interface/t8_fortran_interface_mod.f90

```diff
@@ -0,0 +1,327 @@
+            end function t8_forest_get_tree_num_elements
+      end Interface
+
+      Interface
+            type (c_ptr) function t8_forest_get_element_in_tree (forest, ltreeid, leid_in_tree) &
+                                    bind (c, name = 't8_forest_get_element_in_tree')
+                  use, intrinsic :: ISO_C_BINDING, only: c_ptr, c_int
+                  IMPLICIT NONE
+                  type (c_ptr), value :: forest
+                  integer (c_int), value :: ltreeid, leid_in_tree
+            end function t8_forest_get_element_in_tree
+      end Interface
+
+      Interface
+            subroutine t8_forest_element_from_ref_coords (forest, ltreeid, element, ref_coords, num_coords, coords_out) &
+                                    bind (c, name = 't8_forest_element_from_ref_coords')
+                  use, intrinsic :: ISO_C_BINDING, only: c_ptr, c_int, c_double
+                  IMPLICIT NONE
+                  type (c_ptr), value :: forest, element
+                  integer (c_int), value :: ltreeid, num_coords
+                  real (c_double), dimension(3) :: ref_coords, coords_out
+            end subroutine t8_forest_element_from_ref_coords
+      end Interface
+
+      Interface
+            subroutine t8_global_productionf_noargs_f (string) &
+                                    bind (c, name = 't8_global_productionf_noargs')
+                  use, intrinsic :: ISO_C_BINDING, only: c_char
+                  IMPLICIT NONE
+                  character (c_char) :: string
+            end subroutine t8_global_productionf_noargs_f
+      end Interface
+
+      Interface
+            subroutine t8_fortran_finalize_f () &
+                                    bind (c, name = 't8_fortran_finalize')
+                  IMPLICIT NONE
+            end subroutine t8_fortran_finalize_f
+      end Interface
+
+       Interface 
+             type (c_ptr) function t8_fortran_adapt_by_coordinates_f (forest, recursive, callback) &
+                                     bind (c, name = 't8_fortran_adapt_by_coordinates')
+                   use, intrinsic :: ISO_C_BINDING, only : c_ptr, c_int
+                   IMPLICIT NONE
+                   type (c_ptr), value :: forest
+                   integer (c_int), value :: recursive
+                   type (c_ptr), value :: callback
+             end function t8_fortran_adapt_by_coordinates_f
+       end Interface
+
+      Interface
+            subroutine t8_fortran_element_volume_f (forest, ltreeid, element) & 
+                                    bind (c, name = 't8_forest_element_volume')
+                  use, intrinsic :: ISO_C_BINDING, only: c_ptr, c_int
+                  IMPLICIT NONE
+                  type (c_ptr), value :: forest
+                  integer (c_int), value :: ltreeid
+                  type (c_ptr), value :: element
+            end subroutine t8_fortran_element_volume_f
+      end Interface
+
+End module t8_fortran_interface_mod

```

Comment:
I have no idea about fortran, so I am not able to properly review this file. It looks organized though, and the tests pass.